libraries/package-deps-hash/src/getPackageDeps.ts

```diff
@@ -225,36 +250,43 @@ export function gitStatus(path: string, gitPath?: string): string {
 
 /**
  * Builds an object containing hashes for the files under the specified `packagePath` folder.
- * @param packagePath - The folder path to derive the package dependencies from. This is typically the folder
+ * @param packagePath - The absolute folder path to derive the package dependencies from. This is typically the folder
  *                      containing package.json.  If omitted, the default value is the current working directory.
- * @param excludedPaths - An optional array of file path exclusions. If a file should be omitted from the list
- *                         of dependencies, use this to exclude it.
+ * @param options - Options controlling how the resulting map will be built.
  * @returns the package-deps.json file content
  *
  * @public
  */
 export function getPackageDeps(
   packagePath: string = process.cwd(),
-  excludedPaths?: string[],
-  gitPath?: string
+  options: IGetPackageDepsOptions = {}
 ): Map<string, string> {
-  const gitLsOutput: string = gitLsTree(packagePath, gitPath);
+  const gitLsOutput: string = gitLsTree(packagePath, options.gitPath);
 
   // Add all the checked in hashes
-  const result: Map<string, string> = parseGitLsTree(gitLsOutput);
+  let result: Map<string, string> = parseGitLsTree(gitLsOutput);
+
+  // If requested, transform all paths relative to a new root path
+  if (options.relativeToPath && options.relativeToPath !== packagePath) {
+    const newResult: Map<string, string> = new Map();
+    for (const [key, value] of result.entries()) {
+      newResult.set(path.relative(options.relativeToPath, path.join(packagePath, key)), value);
```

Comment:
This call is both expensive in terms of performance and destroys the previously portable nature of this path (Git always returns '/' as the delimiter)

Reply:
Hence my recommendation that we do the transform to the `projectRelativeFolder` to normalize it to the Git root, then look up everything in terms of Git path.

Reply:
🤔 You're right about the portability, although I could change to `path.posix` to fix that.

I wasn't expecting the performance comment, I was imagining both `path.` calls as essentially string operations under the covers.  (Not much slower than interacting with the Map.)

I guess I would expect that no matter what, we'd want `projectRelativeFolder` to always be relative to `rush.json`, because that's how it would behave if there was no git repo at all.  (I agree with the having package-deps-hash retrieve everything relative to git root, but then I still think you need some kind of speedy way to remap them all to be relative to rush.json).